js/piwik.js

```diff
@@ -7224,6 +7231,7 @@ if (typeof window.Piwik !== 'object') {
              * time the user gives you consent, you do not need to ever call `_paq.push(['setConsentGiven'])`.
              */
             this.requireConsent = function () {
+                configConsentRequired = true;
```

Comment:
shouldn't this be set when `setConsentGiven` is called rather than `requireConsent`?

Reply:
`setConsentGiven` would not be called if `requireConsent` isn't, unless it's for undo-ing opt-out. And in that case I don't think it's explicit consent, it's still assumed consent, right?

Here are the cases I thought of:

user needs to get explicit consent for GDPR:
- user calls requireConsent()
- setConsentGIven called
- &consent=1 added to mark request as having explicitly given consent

user doesn't need consent:
- requests are sent w/o any interruption, &consent=1 is *not* added since it is assumed consent

user has opt out form:
- user sends initial request w/o &consent=1
- visitor opts out, forgetConsentGiven is called, no requests tracked
- visitor opts in, setConsentGiven, requests tracked w/o &consent=1 because user is not required to give consent

I guess the question is, should an opted-in visitor (a visitor who can optionally give their consent) be treated the same as a visitor who is required to give their consent.

Reply:
> I guess the question is, should an opted-in visitor (a visitor who can optionally give their consent) be treated the same as a visitor who is required to give their consent.

 a visitor is either required to give their consent, or their data is tracked by default (and our separate opt-out feature relying on 3rd party cookies applies).